packages/nimble-components/src/rich-text/models/markdown-parser.ts

```diff
@@ -88,10 +95,28 @@ export class RichTextMarkdownParser {
                     // See: https://github.com/ni/nimble/issues/1527
                     excludes: '_',
                     toDOM(node) {
+                        const href = node.attrs.href as string;
+                        const currentMention = RichTextMarkdownParser.mentionConfigs?.find(
+                            mention => mention.isValidMentionHref(href)
+                        );
+                        const displayName = currentMention?.getDisplayName(href);
+
+                        if (currentMention && displayName) {
+                            return [
+                                currentMention.viewElement,
+                                {
+                                    'mention-href': href,
+                                    'mention-label': displayName,
+                                    'disable-editing': true
+                                }
+                            ];
+                        }
+
                         return [
                             anchorTag,
                             {
-                                href: node.attrs.href as Attr,
+                                // Absolute links whose scheme is not HTTP/HTTPS will render as anchor tag without `href` attribute
```

Comment:
I'm not caught up on the discussions around this. What's the rationale for rendering other schemes as an anchor with no `href`? Alternatives might be:
1. rendering them as text as they are today. e.g. by returning a `<div>` or `<span>`
2. rendering them with an href so users can click them. (I think I recall this being a conscious decision to limit the schemes we start out supporting?)

According to [this thread](https://stackoverflow.com/questions/5637969/is-an-empty-href-valid), clicking an anchor with an empty href results in the page reloading which seems undesirable to me. Though when I test this in the browser with an `ftp://ni.com` link I currently see it render as plain text content, so I don't think I understand how to hit this code path.

Reply:
This decision was taken during one of the discussions in the HLD and here is the link to the conversation: https://github.com/ni/nimble/pull/1594#discussion_r1377876092.

The summary of the discussion is all the links in the rich text components should be loaded as `anchor` tags irrespective of their scheme because the absolute link and mention share the same markdown format. So the goal for us is to render mention nodes even if it is not actually a mention in the first place but an absolute link with the matching pattern of mention. 

For instance, initially, if the markdown string contains `<user:1>` but the configuration element is not added, it renders it as plain text (assuming non-HTTPS/HTTP links) in the editor within the `<p>` tag. Now when the configuration element is added dynamically it has the pattern of `user:.*` so it is a mention now. Here when performing getMarkdown + setMarkdown, the string will not be in autolink format as the DOM rendered within the `<p>` node so it won't be converted into a mention node.

To address this scenario, we decided to render all the supported autolink within `anchor` tag so that when getMarkdown is called, we get the string in autolink format and setMarkdown will parse the string and convert the same link to a mention node when configuration element added dynamically.

But non-HTTPS/HTTP links should not be clickable links as per the requirement of absolute links, we decided to remove the `href` as suggested by Milan [here](https://github.com/ni/nimble/pull/1594#discussion_r1379256654) and seems it is a right way to do it.

> I don't think I understand how to hit this code path.

The default behavior of all the HTML `<a>` elements renders as plain text if there is no `href`. But the thread you linked above seems to have an empty string to the `href` attribute but in our case we completely removed the attribute if it is not a HTTPS/HTTP link.

Reply:
Thanks for the explanation, that makes sense. My main concern is ensuring we have test coverage for all these cases. I haven't looked at the spec files yet though.

Reply:
I have written test cases to cover all these cases in the `markdown-parser.spec.ts` file.